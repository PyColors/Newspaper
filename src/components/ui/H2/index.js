import styled from 'styled-components';
import theme from '../../../styles/theme';

const { fontSize } = theme;

const H2 = styled.h2`font-size: ${fontSize.large};`;

export default H2;
