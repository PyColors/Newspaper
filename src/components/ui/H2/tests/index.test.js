import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import H2 from '../../H2';

describe('<H2 />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <H2 class="className">
          {' '}
          <span>Description</span>>
        </H2>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render an <H1> tag', () => {
    const renderedComponent = shallow(<H2 />);
    expect(renderedComponent.type()).toEqual('h2');
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(<H2>{children}</H2>);
    expect(renderedComponent.contains(children)).toBe(true);
  });

  it('should render a prop', () => {
    const id = 'testH2Id';
    const renderedComponent = shallow(<H2 id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });
});
