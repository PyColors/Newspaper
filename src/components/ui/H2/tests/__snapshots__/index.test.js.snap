// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> should match the snapshot 1`] = `
<h2
  class="className"
  className="sc-bdVaJa ikrnGl"
>
   
  <span>
    Description
  </span>
  &gt;
</h2>
`;
